### PR TITLE
draft decisionPropertiesPanel.spec.ts

### PR DESCRIPTION
**Pull request by jomarko**

### CODE REVIEW

File: packages/dmn-editor/tests/e2e/__fixtures__/base.ts [hunk 1]
```diff
@@ -32,6 +37,11 @@ type DmnEditorFixtures = {
   jsonModel: JsonModel;
   nodes: Nodes;
   palette: Palette;
+  generalProperties: GeneralProperties;
+  decisionProperties: DecisionProperties;
+  knowledgeSourceProperties: KnowledgeSourceProperties;
+  propertiesPanel: PropertiesPanelBase;
+  generalDecisionServiceProperties: GeneralDecisionServiceProperties;
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/__fixtures__/base.ts:44]: Looking at this, I don't think a consistency between the names.


```suggestion
  generalPropertiesPanel: GeneralPropertiesPanel;
  decisionPropertiesPanel: DecisionPropertiesPanel;
  knowledgeSourcePropertiesPanel: KnowledgeSourcePropertiesPanel;
  propertiesPanel: PropertiesPanel;
  decisionServicePropertiesPanel: DecisionServicePropertiesPanel;
```

Also, what is `generalProperties`? If we want to use this class as a base for the others, we still need to create the specific ones. After we have the specific, we wouldn't need to directly use the "GeneralProperties" class.

From this, I can see it's missing a few panels:
- TextAnnotation
- Group
- InputData
- BKM
- Global (when no node is selected)
- Multi (when multiple nodes are selected)

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/decisionProperties.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { PropertiesPanelBase } from "./propertiesPanelBase";
+
+export class DecisionProperties extends PropertiesPanelBase {
+  public async changeNodeQuestion(args: { nodeName: string; newQuestion: string }) {
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/decisionProperties.ts:23]: As the panel is from the Node, we don't need to add it to the name. And as we're using the `get`, we could use `set`. This way we would have `setQuestion`. The same applies to the other methods.

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/generalDecisionServiceProperties.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { NodePosition } from "../nodes";
+import { GeneralProperties } from "./generalProperties";
+
+export class GeneralDecisionServiceProperties extends GeneralProperties {
+  public async selectNodeToLoadPropertiesPanel(args: { nodeName: string }) {
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/generalDecisionServiceProperties.ts:24]: I'm not sure we need this at all. The select node is already on the `node` fixture, as you have used.

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/generalProperties.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { DataType } from "../jsonModel";
+import { PropertiesPanelBase } from "./propertiesPanelBase";
+
+export class GeneralProperties extends PropertiesPanelBase {
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/generalProperties.ts:23]: CommonPropertiesPanel?

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/propertiesPanelBase.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export class PropertiesPanelBase {
+  constructor(public diagram: Diagram, public nodes: Nodes, public page: Page) {}
+
+  public panel() {
+    return this.page.getByTestId("properties-panel-container");
+  }
+
+  public async selectNodeToLoadPropertiesPanel(args: { nodeName: string }) {
+    await this.nodes.select({ name: args.nodeName });
+  }
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/propertiesPanelBase.ts:33]: As mentioned before, we don't need this. Selecting the node, from the `nodes` fixture.

File: packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  });
+
+  test("should change the BKM node font - family", async ({ generalProperties }) => {
+    await generalProperties.changeNodeFont({ nodeName: DefaultNodeName.BKM, newFont: "Verdana" });
+
+    expect(await generalProperties.getNodeFont({ nodeName: DefaultNodeName.BKM })).toBe("Verdana");
+  });
+
+  test.skip("should change the BKM node shape - background color", async ({ nodes, propertiesPanel }) => {
+    // blocked https://github.com/microsoft/playwright/issues/19929#issuecomment-1377035969
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts:79]: This issue is related to Firefox, do we have the same problem on chromium and webkit?

File: packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts:1]: `changeBkmProperties`

File: packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  test.beforeEach(async ({ palette, nodes, propertiesPanel }) => {
+    await palette.dragNewNode({ type: NodeType.BKM, targetPosition: { x: 100, y: 100 } });
+    await nodes.select({ name: DefaultNodeName.BKM });
+    await propertiesPanel.open();
+  });
+
+  test("should change the BKM node name", async ({ nodes, generalProperties }) => {
+    await generalProperties.changeNodeName({ from: DefaultNodeName.BKM, to: "Renamed BKM" });
+
+    await expect(nodes.get({ name: "Renamed BKM" })).toBeVisible();
```
Comment (ljmotta) [inline packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts:38]: We need to make an assertion on the properties panel too.

**Review by ljmotta — CHANGES_REQUESTED** [review]
I've added a few comments inline. But the major change would be to add the missing classes and tests, and to use them instead of the `generalProperties` on the tests.

For example on the bkm tests.
`await bkmPropertiesPanel.setName({ ... })` instead of `await generalProperties.changeNodeName({ ... })`

File: packages/dmn-editor/src/propertiesPanel/FontOptions.tsx [hunk 8]
```diff
@@ -375,6 +374,7 @@ export function FontOptions({ startExpanded, nodeIds }: { startExpanded: boolean
             </div>
             <br />
             <Select
+              ouiaId="node-font-style-selector"
```
[the remark this answers is not in the document]
Comment (jomarko) [reply packages/dmn-editor/src/propertiesPanel/FontOptions.tsx:377]: it is used in current `changeNodeFont` and `getNodeFont`
```
  public async changeNodeFont(args: { nodeName: string; newFont: string }) {
    await this.selectNodeToLoadPropertiesPanel({ nodeName: args.nodeName });
    await this.panel().getByTitle("Expand / collapse Font").click();

    await this.panel().locator("[data-ouia-component-id='node-font-style-selector']").click();
    await this.panel().getByText(args.newFont).click();

    await this.diagram.resetFocus();
  }

  public async getNodeFont(args: { nodeName: string }) {
    await this.selectNodeToLoadPropertiesPanel({ nodeName: args.nodeName });
    await this.panel().getByTitle("Expand / collapse Font").click();

    return await this.panel().locator("[data-ouia-component-id='node-font-style-selector']").textContent();
  }
```

[the remark this answers is not in the document]
Comment (ljmotta) [reply packages/dmn-editor/src/propertiesPanel/FontOptions.tsx:377]: Oh ok, it seens I've missed this. We don't use `ouiaId` on any other place. Please use a `data-*` instead. The `ouiaId` is a Patternfly prop.

[the remark this answers is not in the document]
Comment (jomarko) [reply packages/dmn-editor/src/propertiesPanel/FontOptions.tsx:377]: Incorporated as part of #27

File: packages/dmn-editor/tests/e2e/__fixtures__/base.ts (hunk 1 above)
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/__fixtures__/base.ts:44]: Incorporated as part of #27

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/decisionProperties.ts (hunk 2 above)
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/decisionProperties.ts:23]: Incorporated as part of https://github.com/ljmotta/kie-tools/pull/27

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/generalDecisionServiceProperties.ts (hunk 3 above)
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/generalDecisionServiceProperties.ts:24]: Well this was an attempt to minimize code repetition and consequence of class hierarchy we have, I need to rethink more during #27 completion.

File: packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/propertiesPanelBase.ts (hunk 5 above)
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/__fixtures__/propertiesPanel/propertiesPanelBase.ts:33]: Well this was an attempt to minimize code repetition and consequence of class hierarchy we have, I need to rethink more during #27 completion.

File: packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts (hunk 6 above)
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts:79]: I need to recheck as part of #27

File: packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts:1]: done in #27

File: packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts (hunk 7 above)
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeBKMProperties.spec.ts:38]: done in #27

**Comment by jomarko** [comment]
Closing this PR in favor of #27 because both PRs are too interconnected and split if incorporating the feedback to both #26 and #27 would make the work difficult.

**Pull request closed without merging**